### PR TITLE
filtering out unregistered cases

### PR DESCRIPTION
**Pull request by calellowitz**
@sheelio @kaapstorm

### CODE REVIEW

**Review by kaapstorm — APPROVED** [review]

File: custom/icds_reports/ucr/data_sources/ccs_record_cases_monthly_v2.json [hunk 1]
```diff
@@ -1377,6 +1377,43 @@
               }
             },
             "property_value": 183
+          },
+          {
+            "type": "or",
```
Comment (sheelio) [inline custom/icds_reports/ucr/data_sources/ccs_record_cases_monthly_v2.json]: @calellowitz 

Shouldn't this be an "and" (the person is not not_registered AND not migrated)?

Comment (sheelio) [inline custom/icds_reports/ucr/data_sources/ccs_record_cases_monthly_v2.json]: (guessing this should apply wherever you've added this check)

Comment (calellowitz) [inline custom/icds_reports/ucr/data_sources/ccs_record_cases_monthly_v2.json]: I think this is right. It includes them (which is my understanding of a filter, let me know if thats wrong) when the person is NOT `registered_status=not_registered` OR NOT `migrated=yes`, which is to say it includes them if either of the two does not match the exclusion condition. My read of the spec is that both of those properties need to be met to exclude them but it is a bit ambiguous

Comment (sheelio) [inline custom/icds_reports/ucr/data_sources/ccs_record_cases_monthly_v2.json]: @calellowitz I'm pretty sure that the rules are:
"valid" = registered_status != not_registered and migrated != yes

Basically as soon as you're migrated OR as soon as you're marked as "not registered" you're no longer considered a valid beneficiary.  Either should exclude them.

**Comment by calellowitz** [comment]
@sheelio confirmed that you are correct with srikant. have updated the filter

**Review by sheelio — APPROVED** [review]

**Review by snopoke — APPROVED** [review]

**Pull request merged**